modules/fluid_properties/src/fluidproperties/IdealGasMixtureFluidProperties.C

```diff
@@ -0,0 +1,225 @@
+                                                                                                   \
+    Real y = 0;                                                                                    \
+    for (const auto i : make_range(_n_components))                                                 \
+      y += psi[i] * _component_fps[i]->prop##_from_p_T(p, T);                                      \
+                                                                                                   \
+    return y;                                                                                      \
+  }                                                                                                \
+  ADReal IdealGasMixtureFluidProperties::prop##_from_p_T(                                          \
+      const ADReal & p, const ADReal & T, const std::vector<ADReal> & x_secondary) const           \
+  {                                                                                                \
+    const auto x = secondaryToAllMassFractions(x_secondary);                                       \
+    const auto psi = molarFractionsFromMassFractions(x);                                           \
+    mooseAssert(psi.size() == _n_components, "Size mismatch");                                     \
+                                                                                                   \
+    ADReal y = 0;                                                                                  \
+    for (const auto i : make_range(_n_components))                                                 \
+      y += psi[i] * _component_fps[i]->prop##_from_p_T(p, T);                                      \
+                                                                                                   \
+    return y;                                                                                      \
+  }
+
+define_mass_specific_prop_from_p_T(v);
+define_mass_specific_prop_from_p_T(e);
+define_mass_specific_prop_from_p_T(s);
+define_mass_specific_prop_from_p_T(cp);
+define_mass_specific_prop_from_p_T(cv);
+
+define_transport_prop_from_p_T(mu);
+define_transport_prop_from_p_T(k);
+
+#undef define_mass_specific_prop_from_p_T
+#undef define_transport_prop_from_p_T
+
+InputParameters
+IdealGasMixtureFluidProperties::validParams()
+{
+  InputParameters params = VaporMixtureFluidProperties::validParams();
+  params += NaNInterface::validParams();
+
+  params.addClassDescription("Class for fluid properties of an ideal gas mixture");
+
+  params.addRequiredParam<std::vector<UserObjectName>>(
+      "component_fluid_properties",
+      "Name of component fluid properties user objects. The first entry should be the primary "
+      "component.");
+
+  // This is necessary because initialize() must be called before any interface
+  // can be used (which can occur as early as initialization of variables).
```

Comment:
so if it s used for initalization of variables, should it be set to execute on pre_ic ? or is that set automatically when a dependency is found?

Reply:
I don't understand this comment/line. It's copy and paste from Matthias. I'm going to delete it in both this one and the other mixture one because neither of them override `initialize()`, so I can't see why this would be a requirement.

Reply:
deletion works for me